harness/determined/pytorch/_pytorch_context.py

```diff
@@ -92,14 +130,28 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         self._reducers = pytorch._PyTorchReducerContext()
         self._determined_profiler = None  # type: Optional[profiler.ProfilerAgent]
 
-        optimizations_config = self.env.experiment_config.get_optimizations_config()
-        self._aggregation_frequency = cast(int, optimizations_config.get("aggregation_frequency"))
-        self._fp16_compression = cast(bool, optimizations_config.get("gradient_compression"))
-        self._average_aggregated_gradients = cast(
-            bool, optimizations_config.get("average_aggregated_gradients")
-        )
-        self._average_training_metrics = cast(
-            bool, optimizations_config.get("average_training_metrics")
+        self._cluster_info = det.get_cluster_info()
+        self._managed_training = self._cluster_info and self._cluster_info.task_type == "TRIAL"
+
+        self._aggregation_frequency = aggregation_frequency
+        self._fp16_compression = fp16_compression
+        self._average_aggregated_gradients = average_aggregated_gradients
+
+        self._stop_requested = False
+
+    @staticmethod
+    def from_env(env_context: det.EnvContext, core_context: det.core.Context) -> "PyTorchTrialContext":
```

Comment:
no adding public names!

Seriously, please stop making me tell you this.  I don't want my review to be the only layer of protection keeping us from exposing new public APIs on accident.  I need you to pay much more attention to this.

Reply:
got it. it's a big deal, i agree. i guess this is why you recommended doing this from the get-go. my strategy of "just do it all at the end" is clearly lacking since i thought i did a pass at all of these.

Reply:
I understand that it is hard to get all the details right on the first pass, and sometimes there is something you know is wrong but you can't put another thought on your brain's stack to handle it.

In those cases, I use an `# XXX: short explanation for myself` comment to mark the thing as "not right even though I can't think about it right now".

It's nice because it lets me keep focused on whatever other subproblem I was trying to solve when I typed out the wrong code.

That's doubly nice because a lot of editors highlight `XXX` comments, and also we have no `XXX` comments in our whole codebase, so `git grep XXX` will highlight

And it's triply nice because it gets reviewers off your back, since the reviewer doesn't have to leave a comment about the issue, knowing you are already aware and you will fix it.